File: test-suite/src/test/java/io/micronaut/chatbots/telegram/HelloWorldHandler.java
package io.micronaut.chatbots.telegram;

import io.micronaut.chatbots.core.SpaceParser;
import io.micronaut.chatbots.telegram.api.Chat;
import io.micronaut.chatbots.telegram.api.Update;
import io.micronaut.chatbots.telegram.api.send.SendMessage;
import io.micronaut.chatbots.telegram.core.SendMessageUtils;
import io.micronaut.chatbots.telegram.core.TelegramBotConfiguration;
import io.micronaut.chatbots.telegram.core.TelegramHandler;
import io.micronaut.context.annotation.Requires;
import org.jspecify.annotations.Nullable;
import jakarta.inject.Singleton;

import jakarta.validation.constraints.NotNull;
import java.util.Optional;

@Requires(property = "spec.name", value = "telegram")
// tag::class[]
@Singleton
class HelloWorldHandler implements TelegramHandler<SendMessage> {

    private final SpaceParser<Update, Chat> spaceParser;

    HelloWorldHandler(SpaceParser<Update, Chat> spaceParser) {
        this.spaceParser = spaceParser;
    }

    @Override
    public boolean canHandle(@Nullable TelegramBotConfiguration bot, @NotNull Update input) {
        return input.getMessage().getText().contains("hello");
    }

    @Override
    public Optional<SendMessage> handle(@Nullable TelegramBotConfiguration bot, @NotNull Update input) {
        return SendMessageUtils.compose(spaceParser, input, "Hello World");
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/chatbots/docs/telegram/HelloWorldHandler.py
from jakarta.inject import Singleton
from java.util import Optional
from micronaut.chatbots.core import SpaceParser
from micronaut.chatbots.telegram.api import Chat, Update
from micronaut.chatbots.telegram.api.send import SendMessage
from micronaut.chatbots.telegram.core import SendMessageUtils, TelegramBotConfiguration, TelegramHandler
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="telegram")
# tag::class[]
@Singleton
class HelloWorldHandler(TelegramHandler[SendMessage]):

    def __init__(self, space_parser: SpaceParser[Update, Chat]):
        self.space_parser = space_parser

    def canHandle(self, bot: TelegramBotConfiguration | None, input: Update) -> bool:
        return "hello" in input.getMessage().getText()

    def handle(self, bot: TelegramBotConfiguration | None, input: Update) -> Optional[SendMessage]:
        return SendMessageUtils.compose(self.space_parser, input, "Hello World")
# end::class[]
